hashicorp/boundary-ui
# refactor: 💡 remove rose::notification in favor of hds

✅ Closes: https://hashicorp.atlassian.net/browse/ICU-15634

# Description

Replace the rose notification component used by flashMessages service with Hds::Toast.
Hds::Toast guidelines specify [placement of toast](https://helios.hashicorp.design/components/toast#placement) to be bottom right.
Admin UI:
success toast

https://github.com/user-attachments/assets/94e805a5-6136-4e5d-a2da-b56b8579f6dd

error toast

https://github.com/user-attachments/assets/f84e2002-7d5e-4b1c-9abe-1405c97235c8

authentication-complete & authentication-error hds alerts

https://github.com/user-attachments/assets/de724008-f545-41cd-a4c1-901be2814338


Desktop UI:

toast

https://github.com/user-attachments/assets/60097881-629d-441f-825b-1dabda2d57a7





## Screenshots (if appropriate)

## How to Test

1. Go any resource (ex. groups) and create a new one successfully.
2. To see the error toast go to any resource (ex. targets) and cause an error in creating by submitting without required fields.
3. To see authentication alerts go to the http://localhost:4200/authentication-complete & http://localhost:4200/authentication-error in browser
4. In Desktop UI see the toast by cancelling a session.

## Checklist
